aws-lc-rs/src/aead.rs

```diff
@@ -590,6 +611,42 @@ impl LessSafeKey {
         open_within_(&self.key, nonce, aad, in_out, ciphertext_and_tag)
     }
 
+    /// Authenticates and decrypts (“opens”) data into another provided slice.
+    ///
+    /// `aad` is the additional authenticated data (AAD), if any.
+    ///
+    /// On input, `in_ciphertext` must be the ciphertext. The tag must be provided in
+    /// `in_tag`.
+    ///
+    /// The `out_plaintext` length must match the provided `in_ciphertext`.
+    ///
+    /// # Errors
+    /// `error::Unspecified` when ciphertext is invalid. In this case, `out_plaintext` may
+    /// have been overwritten in an unspecified way.
+    ///
+    #[inline]
+    #[allow(clippy::needless_pass_by_value)]
+    pub fn open_separate_gather<A>(
```

Comment:
Is this preferable over allowing the buffer to be in/out, which is supported by [the underlying API](https://github.com/aws/aws-lc/blob/main/include/openssl/aead.h#L381). This also matches "scatter". 

For example:
```rust
    pub fn open_in_place_gather<A>(
        &self,
        nonce: Nonce,
        aad: Aad<A>,
        in_tag: &[u8],
        in_out: &mut [u8],
    ) -> Result<(), Unspecified>
```

Reply:
not sure I follow... this API is specifically for decrypting from one buffer into another. Having an `in_out` means we would need to copy it into the destination buffer, either before or after calling decrypt.

Reply:
Sorry, I wasn't paying attention -- the whole point of this PR is to allow a copy to be avoided.